controllers/glance_controller.go

```diff
@@ -669,8 +669,11 @@ func (r *GlanceReconciler) reconcileNormal(ctx context.Context, instance *glance
 
 	// We reached the end of the Reconcile, update the Ready condition based on
 	// the sub conditions
-	instance.Status.ObservedGeneration = instance.Generation
-	if instance.Status.Conditions.AllSubConditionIsTrue() {
+	subGen, err := r.checkGlanceAPIsGeneration(ctx, instance)
+	if err != nil {
+		return ctrlResult, err
```

Comment:
I might want to enforce a `False` condition here: if I'm here it means I'm not able to get the subCRs and there's a problem in my deployment. Hopefully I already have mirrored the real issue in the top-level `ReadyCondition`, but explicitly marking a condition as False won't hurt. I might add it in a different commit/follow up because the focus is to see if this pattern can be finalized first.